src/solidity-frontend/solidity_convert.cpp

```diff
@@ -900,18 +953,22 @@ bool solidity_convertert::get_binary_operator_expr(
   exprt lhs, rhs;
   if(expr.contains("leftHandSide"))
   {
+    nlohmann::json literalType = expr["leftHandSide"]["typeDescriptions"];
+
     if(get_expr(expr["leftHandSide"], lhs))
       return true;
 
-    if(get_expr(expr["rightHandSide"], rhs))
+    if(get_expr(expr["rightHandSide"], literalType, rhs))
```

Comment:
why are you passing the LHS type with the RHS expression? Shouldn't it be the expr's type?

Reply:
The way the AST is structured is that if the expression is `uint8 i = 22`, this is passed in with the type on the RHS to be just a generic type of int_const without any indication to which variant of the int/uint it is. We make the assumption that the literal on the right will share the type of the rhs. which is why we pass in that literal type.

Reply:
Right, but shouldn't you get the type of expression, instead of the RHS? Something like `expr["typeDescription"]`?